### PR TITLE
jenkins: openssl1.1.0 not supported on node.js 12+

### PR DESCRIPTION
**Pull request by sam-github**
See: https://github.com/nodejs/node/pull/26209#issuecomment-469777736

### CODE REVIEW

**Comment by sam-github** [comment]
nodedjs/master advertises its version as v12.0.0-pre, so I hope this will apply to it.

**Comment by refack** [comment]
made a test job to test this:
https://ci.nodejs.org/job/temp2-node-test-commit-linux-containered/4/console

BTW: we might want to add a test with a shared 1.1.1

**Comment by refack** [comment]
P.S. meanwhile for testing you can trigger the job and un-checking the incompatible configuration.
https://ci.nodejs.org/job/node-test-commit-linux-containered/build?delay=0sec

**Comment by sam-github** [comment]
Sorry, I really don't know enough about jenkins to know how to test this.

I did https://ci.nodejs.org/job/node-test-commit-linux-containered/11123/ from https://github.com/nodejs/build/pull/1711#issuecomment-469891521, unchecking some things that shouldn't be run, but I'm not at all sure what that tested.

Should that have used the groovy changes in this file? What am I looking for? What do I do next to get this landed, or confirm that it does whats intended?

**Comment by refack** [comment]
> What do I do next to get this landed

I'd like another @nodejs/build review.

> confirm that it does whats intended?

This run https://ci.nodejs.org/job/temp2-node-test-commit-linux-containered/4/console seems to verify it excludes as expected:
```
17:46:55 Skipping ubuntu1604_sharedlibs_fips20_x64 for Node.js 12
17:46:55 Skipping ubuntu1604_sharedlibs_openssl102_x64 for Node.js 12
17:46:55 Skipping ubuntu1604_sharedlibs_openssl110_x64 for Node.js 12
```

**Comment by refack** [comment]
Testing https://ci.nodejs.org/job/temp2-node-test-commit-linux-containered/5/console with `v11.x-staging`:
```
18:14:07 Node.js version: 11.11.1
18:14:07 Skipping ubuntu1604_sharedlibs_openssl102_x64 for Node.js 11
18:14:07 Skipping ubuntu1604_sharedlibs_fips20_x64 for Node.js 11
18:14:07 Running nodes
```

**Comment by sam-github** [comment]
I don't have perms to make an openssl-1.1.1 containerd build. It looks like the conditional step for `ubuntu1604_sharedlibs_openssl110_x64` in https://ci.nodejs.org/job/node-test-commit-linux-containered/configure would have to be duplicated (careful following that link, it can cause Chrome to freeze), with basically all the `1.1.0` replaced with `1.1.1`.  The build would only be valid for node.js 11.x and above. Once https://github.com/nodejs/node/pull/26270 lands, it would be valid for 10.x and above.

I can't really tell from https://ci.nodejs.org/computer/test-softlayer-ubuntu1604_sharedlibs_container-x64-5/ what the container is. If can get ssh access to test-softlayer-ubuntu1604_docker-x64-1 I could poke around, and  try to see where openssl1.1.0 is (a docker container, somewhere on the fs, not sure). I suspect an openssl 1.1.1 `untar; Config; make` somewhere parallel to the existing openssl 1.1.0 and 1.0.2 is what is needed.

Does anyone in @nodejs/build want to take a shot at configuring this? If not, I'll ask for access. I'll need a bit of guidance, I've never done any jenkins configuration before.

/cc @mhdawson @miladfarca

**Comment by richardlau** [comment]
> I can't really tell from https://ci.nodejs.org/computer/test-softlayer-ubuntu1604_sharedlibs_container-x64-5/ what the container is. If can get ssh access to test-softlayer-ubuntu1604_docker-x64-1 I could poke around, and try to see where openssl1.1.0 is (a docker container, somewhere on the fs, not sure). I suspect an openssl 1.1.1 `untar; Config; make` somewhere parallel to the existing openssl 1.1.0 and 1.0.2 is what is needed.

I think openssl-1.1.1 is already in the container, based on https://github.com/nodejs/build/blob/d308392fc11b4d9e788cd047524b1dee509a3560/ansible/roles/docker/templates/ubuntu1604_sharedlibs.Dockerfile.j2#L54-L62

> Does anyone in @nodejs/build want to take a shot at configuring this? If not, I'll ask for access. I'll need a bit of guidance, I've never done any jenkins configuration before.

I'm not in @nodejs/build, but wouldn't mind volunteering to configure this if you'd prefer someone with more experience editing Jenkins job configuration. I don't have access to this particular job, but do already have access to the CitGM, node-stresss and node-report jobs.

**Comment by refack** [comment]
> I think openssl-1.1.1 is already in the container, based on

> Does anyone in @nodejs/build want to take a shot at configuring this?

If no new Containers are needed. I'll give this a go.

**Comment by refack** [comment]
Ahh, again it seems like @rvagg already set everything up 🙇 . Just needed to flip a switch.
https://ci.nodejs.org/job/temp2-node-test-commit-linux-containered/6/

BTW, it's set up to run for node >= 9 (linked with shared 1.1.1) it that correct?

**Comment by sam-github** [comment]
Yay! Thanks for the help @refack @rvagg and @richardlau 

Until https://github.com/nodejs/node/pull/26270 lands, the 1.1.1 containerd test should only run for node `> 10`. Once 26270 lands, the version test should be as Rod made it, `> 9`. I don't expect we'll backport openssl 1.1.1 it any further down the LTS levels than 10.x

File: jenkins/scripts/VersionSelectorScript.groovy [hunk 1]
```diff
@@ -57,8 +57,9 @@ def buildExclusions = [
   [ /aix61/,                          anyType,     lt(6)   ],
 
   // Shared libs docker containers -------------------------
-  [ /sharedlibs_openssl111/,          anyType,     lt(9)   ],
+  [ /sharedlibs_openssl111/,          anyType,     lt(10)  ],
```
Comment (refack) [inline jenkins/scripts/VersionSelectorScript.groovy]:
```suggestion
  [ /sharedlibs_openssl111/,          anyType,     lt(11)  ],
```

**Comment by refack** [comment]
https://ci.nodejs.org/job/node-test-commit-linux-containered/11164/
```
09:36:30 Node.js major version: 12
09:36:30 Node.js version: 12.0.0
09:36:30 Skipping ubuntu1604_sharedlibs_openssl102_x64 for Node.js 12
09:36:30 Skipping ubuntu1604_sharedlibs_openssl110_x64 for Node.js 12
09:36:30 Skipping ubuntu1604_sharedlibs_fips20_x64 for Node.js 12
09:36:30 Running nodes
09:36:30 Triggering node-test-commit-linux-containered » ubuntu1604_sharedlibs_debug_x64
09:36:30 Triggering node-test-commit-linux-containered » ubuntu1604_sharedlibs_openssl111_x64
09:36:30 Triggering node-test-commit-linux-containered » ubuntu1604_sharedlibs_withoutssl_x64
09:36:30 Triggering node-test-commit-linux-containered » ubuntu1604_sharedlibs_withoutintl_x64
09:36:30 Triggering node-test-commit-linux-containered » ubuntu1604_sharedlibs_zlib_x64
09:36:30 Triggering node-test-commit-linux-containered » ubuntu1604_sharedlibs_shared_x64
```

https://ci.nodejs.org/job/node-test-commit-linux-containered/11165/
```
09:38:14 Node.js major version: 10
09:38:14 Node.js version: 10.15.4
09:38:14 Skipping ubuntu1604_sharedlibs_fips20_x64 for Node.js 10
09:38:14 Skipping ubuntu1604_sharedlibs_openssl111_x64 for Node.js 10
09:38:14 Skipping ubuntu1604_sharedlibs_openssl102_x64 for Node.js 10
09:38:14 Running nodes
09:38:14 Triggering node-test-commit-linux-containered » ubuntu1604_sharedlibs_withoutintl_x64
09:38:14 Triggering node-test-commit-linux-containered » ubuntu1604_sharedlibs_openssl110_x64
09:38:14 Triggering node-test-commit-linux-containered » ubuntu1604_sharedlibs_zlib_x64
09:38:14 Triggering node-test-commit-linux-containered » ubuntu1604_sharedlibs_debug_x64
09:38:14 Triggering node-test-commit-linux-containered » ubuntu1604_sharedlibs_withoutssl_x64
09:38:14 Triggering node-test-commit-linux-containered » ubuntu1604_sharedlibs_shared_x64
```

**Comment by refack** [comment]
https://ci.nodejs.org/job/node-test-commit-linux-containered/11164/nodes=ubuntu1604_sharedlibs_openssl111_x64/testReport/junit/(root)/test/parallel_test_bootstrap_modules/ 💢 
So disabled for the regular job for now. Still available for stabilization in https://ci.nodejs.org/job/temp2-node-test-commit-linux-containered/

**Comment by richardlau** [comment]
I don't think those test failures are related to the changes here: https://github.com/nodejs/node/pull/26520

**Comment by refack** [comment]
> I don't think those test failures are related to the changes here: [nodejs/node#26520](https://github.com/nodejs/node/pull/26520)

Ok, so we're back in business https://ci.nodejs.org/job/node-test-commit-linux-containered/jobConfigHistory/showDiffFiles?timestamp1=2019-02-28_17-50-31&timestamp2=2019-03-08_10-30-49

**Comment by sam-github** [comment]
Thanks @refack

**Pull request merged**